app/controllers/spree/api/base_controller_decorator.rb

```diff
@@ -0,0 +1,5 @@
+Spree::Api::BaseController.class_eval do
+  def user_for_paper_trail
+    current_spree_user.nil? ? 'Public User' : current_spree_user.id
```

Comment:
The api doesn't use `current_spree_user` and instead does authentication through `current_api_user`